src/System.Linq.Expressions/src/System/Linq/Expressions/Interpreter/LightCompiler.cs

```diff
@@ -2179,6 +2179,25 @@ private void EmitThisForMethodCall(Expression node)
             CompileAddress(node, -1);
         }
 
+        private static bool ShouldWritebackNode(Expression node)
+        {
+            if (!node.Type.GetTypeInfo().IsValueType)
+                return false;
+            switch (node.NodeType)
+            {
+                case ExpressionType.Parameter:
+                case ExpressionType.Call:
+                case ExpressionType.ArrayIndex:
```

Comment:
There is a subtle difference between Convert and Unbox. Convert is pushing a value (like in unbox.any IL opcode) and Unbox pushes a reference (as in unbox opcode).
I wonder if one invokes a mutating member on an unboxed receiver, should it see mutations?  (such code is not possible to write in C# directly).